Rework check-container-logs.rb

kelcecil opened the pull request:
I've made a few adjustments for the check-container-logs.rb check:

* Add an optional parameter to limit the log check to recent failures (-t parameter).
* Refactor to make the protocol parameter unnecessary.
* Remove unnecessary dependencies.

analytically commented:
Awesome, thanks. Is this still backwards compatible?

kelcecil commented:
No. I removed the -p (protocol) parameter as it won't actually do anything.
There's a few things I can do to allow for backwards compatibility:

* I can re-add the -p parameter for backwards compatibility with a note
that it won't actually affect anything.
* I can just create an entirely new check with a note in the readme to use
the new check for API version 1.19 (Docker 1.7 or later).

Any suggestions or preference?

On Thu, Dec 3, 2015 at 4:42 AM, Mathias Bogaert <notifications@github.com>
wrote:

> Awesome, thanks. Is this still backwards compatible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sensu-plugins/sensu-plugins-docker/pull/13#issuecomment-161569011>
> .
>

kelcecil commented:
I'll take care of those Rubicop issues as well.

sstarcher commented:
any chance we can get this merged?

analytically commented:
You're welcome to add the other changes here as well.

kelcecil commented:
Will do.

On Tue, Dec 15, 2015 at 11:12 AM Mathias Bogaert <notifications@github.com>
wrote:

> You're welcome to add the other changes here as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sensu-plugins/sensu-plugins-docker/pull/13#issuecomment-164811849>
> .
>

eheydrick commented:
Where are we at with getting this ready for merge? I'd like to get this into the next release.

sstarcher commented:
@kelcecil any updates on this?

kelcecil commented:
@eheydrick @analytically  Sorry for the wait. Dropped the ball on this over the holidays and got side tracked with some over stuff at the beginning of this year.

I've updated my log-rework branch to communicate with docker consistently for each of the check*.rb checks. I haven't touched the metrics*.rb collectors as I have no way to test any changes. Everything now uses -h to specify the host. No need to specify the protocol. The arguments are mostly the same otherwise. Everything passes rubocop without disabling checks.

I tested everything by building and instalingl the gem. Everything seems to work for me, but a second look at that would be helpful.

sstarcher commented:
@kelcecil any reason you moved away from the docker ruby gem and are talking to the docker daemon directly?

kelcecil commented:
@sstarcher There's two gems that were in use: docker and docker-api. The docker gem hadn't been updated since 2013 until this past January (https://rubygems.org/gems/docker). I originally thought the project was abandoned, and the purpose of it was to wrap the CLI tool which could potentially be brittle. The docker-api gem (https://github.com/swipely/docker-api) strives for compatibility with API version 1.15 (Docker 1.3), and I needed a feature (since parameter in the API call for container logs) from API version 1.19 (Docker 1.7).
